core/src/main/java/org/infinispan/BaseCacheStream.java

```diff
@@ -0,0 +1,120 @@
+    * <p>This value is <b>always</b> ignored when this stream is backed by a cache that is not distributed as all
+    * values are already local.</p>
+    * @param batchSize The size of each batch.  This defaults to the state transfer chunk size.
+    * @return a stream with the batch size updated
+    */
+   BaseCacheStream distributedBatchSize(int batchSize);
+
+   /**
+    * Allows registration of a segment completion listener that is notified when a segment has completed
+    * processing.  If the terminal operator has a short circuit this listener may never be called.
+    * <p>This method is designed for the sole purpose of use with the {@link CacheStream#iterator()} to allow for
+    * a user to track completion of segments as they are returned from the iterator.  Behavior of other methods
+    * is not specified.  Please see {@link CacheStream#iterator()} for more information.</p>
+    * <p>Multiple listeners may be registered upon multiple invocations of this method.  The ordering of notified
+    * listeners is not specified.</p>
+    * @param listener The listener that will be called back as segments are completed.
+    * @return a stream with the listener registered.
+    */
+   BaseCacheStream segmentCompletionListener(SegmentCompletionListener listener);
+
+   /**
+    * Disables tracking of rehash events that could occur to the underlying cache.  If a rehash event occurs while
+    * a terminal operation is being performed it is possible for some values that are in the cache to not be found.
+    * Note that you will never have an entry duplicated when rehash awareness is disabled, only lost values.
+    * <p>Most terminal operations will run faster with rehash awareness disabled even without a rehash occuring.
+    * However if a rehash occurs with this disabled be prepared to possibly receive only a subset of values.</p>
+    * @return a stream with rehash awareness disabled.
+    */
+   BaseCacheStream disableRehashAware();
+
+   /**
+    * Sets a given time to wait for a remote operation to respond by.  This timeout does nothing if the terminal
+    * operation does not go remote.
+    * <p>If a timeout does occur then a {@link java.util.concurrent.TimeoutException} is thrown from the terminal
+    * operation invoking thread or on the next call to the {@link Iterator} or {@link Spliterator}.</p>
+    * <p>Note that if a rehash occurs this timeout value is reset for the subsequent retry if rehash aware is
+    * enabled.</p>
+    * @param timeout the maximum time to wait
+    * @param unit the time unit of the timeout argument
+    * @return a stream with the timeout set
+    */
+   BaseCacheStream timeout(long timeout, TimeUnit unit);
+
+   /**
+    * Functional interface that is used as a callback when segments are completed.  Please see
+    * {@link BaseCacheStream#segmentCompletionListener(SegmentCompletionListener)} for more details.
+    * @since 8.0
```

Comment:
Technically it was an interface in 8.0 but it was moved in 9.0, haha.  I can update it.